src/perturb.jl

```diff
@@ -40,57 +38,43 @@ function create_perturbation_gates(s, k, B_pert, α, x, L, N_sites, energy_sign,
 
             # add perturbation via one-body oscillation term to the Hamiltonian
-            # hj = (1/(N_sites-1))* (
-            # ((ω_pert[i] * B[1] * op("Sx", s_i)* op("Id", s_j))  + (ω_pert[j] * B[1] * op("Sx", s_j) * op("Id", s_i))) + 
-            # ((ω_pert[i] * B[2] * op("Sy", s_i)* op("Id", s_j))  + (ω_pert[j] * B[2] * op("Sy", s_j) * op("Id", s_i))) +
-            # ((ω_pert[i] * B[3] * op("Sz", s_i)* op("Id", s_j))  + (ω_pert[j] * B[3] * op("Sz", s_j) * op("Id", s_i))) 
-            # )
-            hj= (1/(N_sites-1))* (
+            hj= (
                 (ω_pert[i] * B_pert[1] * op("Sx", s_i)* op("Id", s_j))  + (ω_pert[i] * B_pert[2] * op("Sy", s_i)* op("Id", s_j))  + (ω_pert[i] * B_pert[3] * op("Sz", s_i)* op("Id", s_j)) )
-                # println("hj1= ", hj1) 
-            hj += (1/(N_sites-1))* (
-                (ω_pert[j] * B_pert[1] * op("Id", s_i) * op("Sx", s_j)) + (ω_pert[j] * B_pert[2]  * op("Id", s_i)* op("Sy", s_j)) + (ω_pert[j] * B_pert[3]  * op("Id", s_i)* op("Sz", s_j)) )
-                # println("hj2= ", hj2)
-            
 
-            #println("hj= ", hj)
             # make Trotter gate Gj that would correspond to each gate in the gate array of ITensors             
-            Gj = exp(-im * τ/2 * hj*  1/hbar)
-            # Gj = exp(-im * τ/2 * hj)
+            Gj = exp(-im * τ/2 * hj*  1/hbar) #Gj has factor of 1/hbar sinceonly Richers inhomo tests need perturbation
+
             # The push! function adds (appends) an element to the end of an array;
             # ! performs an operation without creating a new object, (in a way overwites the previous array in consideration); 
             # i.e. we append a new element Gj (which is an ITensor object representing a gate) to the end of the gates array.
             push!(gates, Gj)
-            #println("perturb_gates= ", gates)
         end
     end
 
     # append! adds all the elements of a gates in reverse order (i.e. (N,N-1),(N-1,N-2),...) to the end of gates array.
     # appending reverse gates to create a second-order Trotter-Suzuki integration
-    append!(gates, reverse(gates))
-    #println(gates)  
+    append!(gates, reverse(gates)) 
     return gates
 end
 
 
 function evolve_perturbation(s,k, τ_pert, B_pert, α, x, L, N_sites, ψ, cutoff, maxdim, energy_sign,ttotal)
 
     # extract the gates array generated in the gates_function file
-    perturb_gates = create_perturbation_gates(s,k, B_pert, α, x, L, N_sites, energy_sign, τ)
+    perturb_gates = create_perturbation_gates(s,k, B_pert, α, x, L, N_sites, energy_sign, τ_pert)
 
      # Compute and print survival probability (found from <Sz>) at each time step then apply the gates to go to the next time
-     for t in 0.0:τ:ttotal  #perhaps not perturn till the end? stop somewhere in the mid 
+     for t in 0.0:τ_pert:ttotal  #perhaps not perturn till the end? stop somewhere in the mid 
```

Comment:
The perturbaton only needs to be applied once - I think we need to discuss this again